assert/assertions.go

```diff
@@ -1174,6 +1174,46 @@ func formatListDiff(listA, listB interface{}, extraA, extraB []interface{}) stri
 	return msg.String()
 }
 
+// NoElementsMatch asserts that the specified listA(array, slice...)
+// doesn't have any elements in common with the listB(array, slice...).
+//
+// assert.NoElementsMatch(t, [1, 2, 3], [4, 5, 6]) -> true
+//
+// assert.NoElementsMatch(t, [1, 2, 3], [1, 4, 5]) -> false
+func NoElementsMatch(t TestingT, listA, listB interface{}, msgAndArgs ...interface{}) (ok bool) {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+	if isEmpty(listA) || isEmpty(listB) {
+		return true
+	}
+
+	if !isList(t, listA, msgAndArgs...) || !isList(t, listB, msgAndArgs...) {
+		return false
+	}
+
+	aValue := reflect.ValueOf(listA)
+	bValue := reflect.ValueOf(listB)
+
+	aLen := aValue.Len()
+	bLen := bValue.Len()
+
+	memo := make(map[interface{}]bool, 0)
+	for i := 0; i < aLen; i++ {
+		element := aValue.Index(i).Interface()
+		memo[element] = true
+	}
+
+	for j := 0; j < bLen; j++ {
+		element := bValue.Index(j).Interface()
+		if _, found := memo[element]; found {
+			return Fail(t, fmt.Sprintf("Found common element %#v in both lists", element), msgAndArgs...)
+		}
+	}
```

Comment:
I'd suggest either make this more efficient with:
```suggestion
	memo := make(map[interface{}]struct{}, 0)
	for i := 0; i < aLen; i++ {
		element := aValue.Index(i).Interface()
		memo[element] = struct{}{}
	}

	for j := 0; j < bLen; j++ {
		element := bValue.Index(j).Interface()
		if _, found := memo[element]; found {
			return Fail(t, fmt.Sprintf("Found common element %#v in both lists", element), msgAndArgs...)
		}
	}
```

Or utilise the boolean with:
```suggestion
	memo := make(map[interface{}]bool, 0)
	for i := 0; i < aLen; i++ {
		element := aValue.Index(i).Interface()
		memo[element] = true
	}

	for j := 0; j < bLen; j++ {
		element := bValue.Index(j).Interface()
		if memo[element] {
			return Fail(t, fmt.Sprintf("Found common element %#v in both lists", element), msgAndArgs...)
		}
	}
```